packages/vitest/src/node/core.ts

```
@@ -390,6 +391,12 @@ export class Vitest {
       await this.globTestFiles(filters),
     )
 
+    if (!this.config.includeTaskLocation && files.some(spec => spec.testLocations)) {
```

This should be an actual `throw new Error` instead of a warning